src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3213,6 +3217,25 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                                          pptr, nullptr);
     }
 
+    ASR::asr_t* create_PointerToCPtr(const AST::Call_t& x) {
+        if( x.n_args != 2 ) {
+            throw SemanticError("p_c_pointer accepts two positional arguments, "
+                                "first a Pointer variable, second a CPtr variable.",
+                                x.base.base.loc);
+        }
+        visit_expr(*x.m_args[0]);
+        ASR::expr_t* arg = ASRUtils::EXPR(tmp);
+        visit_expr(*x.m_args[1]);
+        ASR::expr_t* cptr = ASRUtils::EXPR(tmp);
+        ASR::ttype_t *type = ASR::down_cast<ASR::ttype_t>(ASR::make_Pointer_t(al, x.base.base.loc,
+            ASRUtils::expr_type(arg)));
+        ASR::expr_t* pptr = ASR::down_cast<ASR::expr_t>(ASR::make_GetPointer_t(al, x.base.base.loc, arg, type, nullptr));
```

Comment:
We shouldn't add `GetPointer` on our own. Instead a user should do, `pointer(x)` as the first argument. If `x` already a pointer then no need to a call to `pointer`, just pass `x` as the first argument.

Reply:
Ok, go ahead and submit a PR fixing this, and the test.

Reply:
Note also that we need a corresponding behavior in CPython, and I think it might not be possible with the current Python API, we might need some changes.